### PR TITLE
gravel/glass: mark deployment ready once we finish deploying services

### LINKED ISSUE #205: gravel: /api/status does not report 'ready' stage
After bootstrapping and deploying a CephFS service the `/api/status` endpoint always returns `bootstrapped` in `deployment_state.stage`. At some point the `ready` stage is expected to be returned.

### PR DESCRIPTION
**Pull request by jecluis**
Once we finish deploying services, ensure we mark the deployment as finished.

Resolves #205 

Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: src/gravel/controllers/bootstrap.py [hunk 1]
```diff
@@ -77,6 +77,15 @@ async def bootstrap(self) -> bool:
 
         return True
 
+    async def mark_finished(self) -> bool:
+        stage: DeploymentStage = gstate.config.deployment_state.stage
+        if stage == DeploymentStage.ready:
+            return True
+        elif stage != DeploymentStage.bootstrapped:
+            return False
```
Comment (mgfritch) [inline src/gravel/controllers/bootstrap.py:85]: it feels as if `DeploymentStage.ready` really is the same as `DeploymentStage.bootstrapped` ...

File: src/gravel/controllers/bootstrap.py [hunk 2]
```diff
@@ -77,6 +77,15 @@ async def bootstrap(self) -> bool:
...
+        gstate.config.set_deployment_stage(DeploymentStage.ready)
```
Comment (mgfritch) [inline src/gravel/controllers/bootstrap.py:86]: is it really necessary to have the web client set the ready flag? ... maybe it should be set at the end of a successful bootstrap?

File: src/gravel/controllers/bootstrap.py (hunk 1 above)
Comment (jecluis) [reply src/gravel/controllers/bootstrap.py:85]: it isn't though - bootstrapped means we can now start deploying services, whereas 'ready' means we have gone through the whole installer.

File: src/gravel/controllers/bootstrap.py (hunk 2 above)
Comment (jecluis) [reply src/gravel/controllers/bootstrap.py:86]: and then we have no idea if we have already went through the whole installer. :shrug:

Comment (jhesketh) [reply src/gravel/controllers/bootstrap.py:86]: Maybe we need to separate the concept of deployment_stage from "wizard/installer stage"?

**Comment by jecluis** [comment]
@votdev can you check if this makes sense to you?

**Review by votdev — APPROVED** [review]

**Pull request merged**